### PR TITLE
Add !vc command to manage temporary voice channels

### PR DESCRIPTION
**Pull request by mythmon**
This adds a command, `!vc`, that allows all users to create temporary voice channels, marked with ⚡. Users can delete these channels manually if they want, and there is a command to delete everything made by the plugin. Additionally, channels that go unoccupied for some time (about 10 minutes) will be automatically cleaned up.

This PR still needs tests, but should be good to go besides that.

### CODE REVIEW

**Review by Errolyn — CHANGES_REQUESTED** [review]
This is looking really good! This is going to be a really nice way to trim down our voice channels :D 

A couple of things:

- Should a member be able to add emojis to the voice channel names? 
    - Some emojis work others show up as <:emoji_name:123412341234> inside of the voice channel name.
    - If someone adds the cleanup emoji to the start of the voice channel name it looks like the bot is not updating those channels correctly?
`!vc create :scissors: test`
`!vc create :scissors: test 2`

were used to create these channels:

image

I also created channels with other emojis like:

`!vc create :Rainbow_Weeb: test 3`
`!vc create :pinkheart: test 4`

those were deleted as expected but `:scissors: test 2` has not been deleted though the others have. 

Another thing I encountered:

After all channels are deleted, the bot removes the auto channel group. If you try to add a new voice channel after it has been cleared I am getting an error:

```
DiscordRESTError [50035]: Invalid Form Body
  parent_id: Category does not exist
    at RequestHandler.request (/mnt/c/dev/timelord/node_modules/eris/lib/rest/RequestHandler.js:71:15)
    at BotWrapper.createChannel (/mnt/c/dev/timelord/node_modules/eris/lib/Client.js:432:36)
    at VcCommand.subcommandCreate (/mnt/c/dev/timelord/commands/voiceChannelManager.js:115:20)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

```
From the members perspective they type the command and nothing happens.

---

While sitting in a voice channel and expecting a update to the channels I got this in console but subsequent updates to the channels were fine?

```
(node:1069) UnhandledPromiseRejectionWarning: TypeError: channel.editChannel is not a function
    at VcCommand.cleanupTask (/mnt/c/dev/timelord/commands/voiceChannelManager.js:212:25)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:1069) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:1069) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

File: commands/voiceChannelManager.js [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    if (!this._targetGroup) {
+      const channels = await this.bot.getRESTGuildChannels(guild.id);
+      let targetGroup = channels.find((channel) => channel.name.startsWith(EMOJIS.CHANNEL_PREFIX));
+      if (!targetGroup) {
+        targetGroup = await this.bot.createChannel(
+          guild.id,
+          `${EMOJIS.CHANNEL_PREFIX} AUTO CHANNELS`,
+          CHANNEL_TYPE.GROUP,
+          {
+            reason: `To hold !vc command channesl`,
```
Comment (Errolyn) [inline commands/voiceChannelManager.js:87]:
```suggestion
            reason: `To hold !vc command channels`,
```

File: commands/voiceChannelManager.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+          },
+        );
+      }
+      this._targetGroup = targetGroup;
+    }
+    return this._targetGroup;
+  }
+
+  async findChannel(guild, needle) {
```
Comment (Errolyn) [inline commands/voiceChannelManager.js:96]: needle? as in needle in a hay stack?

Comment (mythmon) [reply commands/voiceChannelManager.js:96]: Yep, that's exactly the metaphor. I spent a while working on some search-engine-like code, and picked it up there.

**Comment by mythmon** [comment]
I fixed the bug when the scissors emoji is included in the name, and I'm inclined to call "weird emojis show up as weird emoji codes" a wont-fix. I don't think Discord can have those emojis in channel names. It would be nice to sanitize them out but... I don't want to go down that rabbit hole of edge cases.

I found a way to reproduce the "Invalid Form Body" error, and I'm looking into that.

**Review by Errolyn — COMMENTED** [review]
looks good, r+ with tests

**Review by Errolyn — APPROVED** [review]
The tests look great! I'm excited to get this up on the server :D
